File: BDD_With_POM/BDD_With_POM/BDD_With_POM/Site Pages/Navigator.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DiscernAI.Site_Pages
{
    public class Navigator
    {
        IWebDriver driver;

        public Navigator(IWebDriver driver)
        {
            this.driver = driver;
        }

        By navLinks = By.CssSelector("mat-accordion a");

        public void ClickNavigationLink(string linkText)
        {
            // Find all anchor elements within mat-accordion
            IList<IWebElement> navLinkClick = driver.FindElements(navLinks);

            // Iterate through the anchor elements
            for (int i = 0; i < navLinkClick.Count; i++)
            {
                // Get the text of the current anchor element
                string currentLinkText = navLinkClick[i].Text.Trim();

                // Check if the link text matches the desired linkText
                if (currentLinkText == linkText)
                {
                    // Click the link if found and return
                    navLinkClick[i].Click();
                    return;
                }
            }
            throw new NoSuchElementException($"Navigation link with text '{linkText}' not found.");
        }
    }
}

File: SmallWorldAutomation/src/test/java/pages/Navigator.java
package pages;

import org.openqa.selenium.By;
import utility.BrowserDriver;
import org.openqa.selenium.WebElement;
import java.util.List;
import org.openqa.selenium.NoSuchElementException;

public class Navigator extends BrowserDriver {

    public static By openHamburger = By.xpath("//button[@id='react-burger-menu-btn'][.='Open Menu']\n");
    public static void OpenHamburgerClick(){
        driver.findElement(openHamburger).click();
    }
    public static By menuLinkLists = By.xpath("//div[@class='bm-menu']//nav[@class='bm-item-list']//a");
    public static void ClickOnMenu(String linkText){

        // Find all anchor elements within the specified XPath
        List<WebElement> navLinkClick = driver.findElements(menuLinkLists);

        // Iterate through the anchor elements
        for (int i = 0; i < navLinkClick.size(); i++) {
            // Get the text of the current anchor element
            String currentLinkText = navLinkClick.get(i).getText().trim();

            // Check if the link text matches the desired linkText
            if (currentLinkText.equals(linkText)) {
                // Click the link if found and return
                navLinkClick.get(i).click();
                return;
            }
        }
        throw new NoSuchElementException("Navigation link with text '" + linkText + "' not found.");
    }
}

